### PR TITLE
Document indexing by different numbers of indices

### PR DESCRIPTION
**Pull request by mbauman**
and remove/supercede an outdated section on the same in devdocs/subarray

### CODE REVIEW

File: doc/src/manual/arrays.md [hunk 1]
```diff
@@ -648,6 +637,118 @@ julia> x[mask]
...
+
+julia> A[5]
+7
+
+julia> vec(A)[5]
+7
+```
+
+A linear index into the array `A` can be converted to a `CartesianIndex` for cartesian
+indexing with [`CartesianIndices(A)[i]`](@ref), and a set of `N` cartesian indices can be
```
Comment (mbauman) [inline doc/src/manual/arrays.md]: Is this valid `@ref` syntax?

File: doc/src/manual/arrays.md [hunk 2]
```diff
@@ -648,6 +637,118 @@ julia> x[mask]
...
+julia> A[1, 3] # Attempts to omit dimensions 3 & 4 (lengths 2 and 1)
+ERROR: BoundsError: attempt to access 3×4×2×1 reshape(::UnitRange{Int64}, 3, 4, 2, 1) with eltype Int64 at index [1, 3]
+
+julia> A[19] # Linear indexing
+19
+```
+
+When omitting _all_ indices with `A[]`, this semantic provides a simple idiom
+to retrieve the only element in an array and simultaneously ensure that there
+only was only one element with.
```
Comment (innerlee) [inline doc/src/manual/arrays.md]:
```suggestion
was only one element.
```

Comment (mbauman) [reply doc/src/manual/arrays.md]: Thanks!

File: doc/src/manual/arrays.md (hunk 1 above)
Comment (fredrikekre) [reply doc/src/manual/arrays.md]: No, worked around it.

**Comment by mbauman** [comment]
Thanks for carrying this across the finish line @fredrikekre!

**Pull request merged**